pkg/controller/infrastructure/templates/main.tpl.tf

```diff
@@ -122,6 +122,16 @@ resource "alicloud_security_group_rule" "allow_all_internal_udp_in" {
   cidr_ip           = "{{ .vpc.cidr }}"
 }
 
+resource "alicloud_security_group_rule" "allow_all_internal_pod_traffic_in" {
+  type              = "ingress"
+  ip_protocol       = "all"
+  policy            = "accept"
+  port_range        = "-1/-1"
+  priority          = 1
+  security_group_id = alicloud_security_group.sg.id
+  cidr_ip           = "{{ .podCIDR }}"
+}
+
```

Comment:
Hi,
this security group rule will be created no matter user uses overlay network or not.  is there any harm?

is there a way for user to disable overlay network for existing shoot cluster?  If yes, we also need to consider how to trigger infrastructure reconlication in order to get this security rule created.

Reply:
Hi,

this security group rule will allow packets in the pod ip range to flow through the virtual network. In case an overlay network is used this rule allows packets, which will never be created. Hence, it is not required in this case, but it also does not harm as on the kubernetes nodes there are anyway local routing rules in place preventing packets in the pod ip range to go out directly if an overlay network is used.
So in short: From my perspective, there is no harm.

It was possible all the time to disable the overlay network in the configuration of the networking extension. However, this would have led to failing cluster in the past. Hence, I do not see an urgent need to add a reconcile flow now. Switching the overlay network is an incompatible operation causing at least some downtime in terms of network connectivity. We are not yet at the stage that we can support this. Therefore, I would not necessarily do it for old clusters, but only for new ones.